packages/swap/src/routes/quote.ts

```diff
@@ -110,41 +113,43 @@ const quote = (io: Server) => {
         const bestQuote = findBestQuote(marketMakerQuotes, brokerQuote);
 
         const quoteSwapFees = await calculateIncludedSwapFees(
-          quoteRequest.source_asset,
-          quoteRequest.destination_asset,
-          quoteRequest.deposit_amount,
+          getInternalAsset(srcAssetChain),
+          getInternalAsset(destAssetChain),
+          String(swapInputAmount),
           'intermediateAmount' in bestQuote
             ? bestQuote.intermediateAmount
             : undefined,
-          bestQuote.egressAmount,
+          bestQuote.outputAmount,
         );
         includedFees.push(...quoteSwapFees);
 
         const egressFee = bigintMin(
-          await getEgressFee(query.destAsset),
-          BigInt(bestQuote.egressAmount),
+          await getEgressFee(destAssetChain),
+          BigInt(bestQuote.outputAmount),
         );
         includedFees.push({
           type: 'EGRESS',
-          chain: query.destAsset.chain,
-          asset: query.destAsset.asset,
+          chain: destAssetChain.chain,
+          asset: destAssetChain.asset,
           amount: egressFee.toString(),
         });
 
-        const egressAmount = BigInt(bestQuote.egressAmount) - egressFee;
+        const egressAmount = BigInt(bestQuote.outputAmount) - egressFee;
 
-        const minimumEgressAmount = await getMinimumEgressAmount(
-          query.srcAsset,
-        );
+        const minimumEgressAmount = await getMinimumEgressAmount(srcAssetChain);
 
         if (egressAmount < minimumEgressAmount) {
           throw ServiceError.badRequest(
             `egress amount (${egressAmount}) is lower than minimum egress amount (${minimumEgressAmount})`,
           );
         }
-        // We want to remove the id from the final response body as it is just a random uuid
-        // eslint-disable-next-line @typescript-eslint/no-unused-vars
-        const { id = undefined, ...response } = {
+
+        // remove the id and output amount from the final response body as it is internal information
+        const {
+          id = undefined, // eslint-disable-line @typescript-eslint/no-unused-vars
+          outputAmount,
+          ...response
+        } = {
           ...bestQuote,
           egressAmount: egressAmount.toString(),
```

Comment:
i would prefer not changing it if we dont have a important reason because its probably one of the types every consumer depends on. changing the prop name would mean all integrators need to change their integration